internal/featuregates/featuregates.go

```diff
@@ -0,0 +1,47 @@
+package featuregates
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/rest"
+
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// CheckAdmissionWebhookMatchConditions returns true if the feature gate
+// AdmissionWebhookMatchConditions is activated. It does this by fetching the
+// OpenAPIV3 schema from the discovery client and checking for the feature
+// gate. This feature is stable since Kubernetes v1.30.
+func CheckAdmissionWebhookMatchConditions(config *rest.Config) (bool, error) {
+	// Obtain openAPIV3 client from discoveryClient
+	apiClient := discovery.NewDiscoveryClientForConfigOrDie(config).OpenAPIV3()
+	paths, err := apiClient.Paths()
+	if err != nil {
+		log.Fatalf("Failed to fetch OpenAPI spec: %v", err)
```

Comment:
why `log.Fatalf` here? couldn't we just return an error?

Reply:
True, rebased and changed.